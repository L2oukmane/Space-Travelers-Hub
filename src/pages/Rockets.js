import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../components/Rocket';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);

  return (
    <div className="rockets d-col-flex">
      {
        rockets.length > 0
          ? rockets.map((rocket) => (
            <Rocket
              key={rocket.id}
              rocket={rocket}
            />
          ))
          : ''
      }
    </div>
  );
};

export default Rockets;