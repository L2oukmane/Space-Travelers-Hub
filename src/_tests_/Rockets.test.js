import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../pages/Rockets';
import testStore from '../redux/configureStore';

const rocket = [
  {
    desc: 'Falcon 9 is a two-stage rocket designed and manufactured by SpaceX',
    key: 'special',
    id: 1,
    name: 'defines the rocket',
    reservation: true,
  },
];

describe('testing the rendering of rockets', () => {
  it('', () => {
    const { container } = render(
      <Provider store={testStore}>
        <Rockets key={rocket[0]} rocket={rocket[0]} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});