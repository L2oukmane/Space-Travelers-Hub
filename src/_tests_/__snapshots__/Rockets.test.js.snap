// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of rockets  1`] = `
<div>
  <div
    class="rockets d-col-flex"
  />
</div>
`;